Added JUnit Test Cases for Replacement

surajdm123 opened the pull request:
### Changes

This PR improves the reliability of `Replacement` component by validating edge-case formatting scenarios.

**Before:**
<img width="938" height="315" alt="Screenshot 2025-07-16 at 10 26 05 PM" src="https://github.com/user-attachments/assets/47a08af5-b240-40ba-8abf-9f20dce46c22" />

**After:**
<img width="947" height="315" alt="Screenshot 2025-07-16 at 10 24 41 PM" src="https://github.com/user-attachments/assets/2a18ffca-0b75-4e18-b071-7246c438ce29" />


### Test Plan

> mvn -Dtest=ReplacementTest test

```
...
[INFO] Running com.google.googlejavaformat.java.ReplacementTest
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.068 s - in com.google.googlejavaformat.java.ReplacementTest
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
```
----

> mvn test

```
[INFO] Results:
[INFO] 
[INFO] Tests run: 1533, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  12.168 s
[INFO] Finished at: 2025-07-16T22:30:47-07:00
[INFO] ------------------------------------------------------------------------

```

### Next Steps

- Add more missing test cases to make this codebase more resilient.

surajdm123 committed:
Added JUnit Test Cases for Replacement

Pankraz76 commented:
sorry no offence given. Nobody perfect of course does rewrite has a lot of bug, but as its the de facto standard its about how you tread them.

Not expecting this kind of absence from a major player like google or even consider palantir. but again no front given. Just my hasty and harsh 2 cents.

Pankraz76 commented:
sorry my not being nice and patient it seems all fine. Maybe just vacay or something. 

Please excuse and enjoy.

cushon reviewed (changes requested):
@surajdm123 can you say more about the goal here, and why you decided to add these specific tests? There is some indirect coverage of `Replacement` from other tests, did you find issues with it you'd like to change as a follow-up?

surajdm123 committed:
Addressed comments for ReplacementTest

surajdm123 commented:
> @surajdm123 can you say more about the goal here, and why you decided to add these specific tests? There is some indirect coverage of `Replacement` from other tests, did you find issues with it you'd like to change as a follow-up?

@cushon 
Thank you so much for reviewing the code and providing valuable feedback. 
I noticed that the Replacement class was missing some test coverage, so I thought of adding tests for those missing lines of code.


**Before:**
<img width="938" height="315" alt="Screenshot 2025-07-16 at 10 26 05 PM" src="https://github.com/user-attachments/assets/47a08af5-b240-40ba-8abf-9f20dce46c22" />

**After:**
<img width="947" height="315" alt="Screenshot 2025-07-16 at 10 24 41 PM" src="https://github.com/user-attachments/assets/2a18ffca-0b75-4e18-b071-7246c438ce29" />

surajdm123 commented:
@cushon 
I've addressed all the feedback in the latest commit. Whenever you have a moment, I'd appreciate it if you could take another look.
Thank you again for your time and insightful feedback!
